include/rocksdb/compaction_filter.h

```diff
@@ -214,16 +210,28 @@ class CompactionFilter {
   }
 };
 
-// Each compaction will create a new CompactionFilter allowing the
-// application to know about different compactions
+// Each thread of work involving creating table files will create a new
+// `CompactionFilter` according to `ShouldFilterTableFileCreation()`. This
+// allows the application to know about the different ongoing threads of work
+// and makes it unnecessary for `CompactionFilter` to provide thread-safety.
 class CompactionFilterFactory {
  public:
   virtual ~CompactionFilterFactory() {}
 
+  // Returns whether a thread creating table files for the specified `reason`
+  // should invoke `CreateCompactionFilter()` and pass KVs through the returned
+  // filter.
+  virtual bool ShouldFilterTableFileCreation(
+      TableFileCreationReason reason) const {
+    // Default implementation only applies `CompactionFilter` to files generated
+    // by compaction, as the name implies.
```

Comment:
I would say or add "for backward compatibility"

Reply:
Done.